application/helpers/common_helper.php

```diff
@@ -1379,7 +1379,7 @@ function createFieldMap($survey, $style = 'short', $force_refresh = false, $ques
             $fieldmap["token"]['group_name'] = "";
         }
     }
-    if ($prow['datestamp'] == "Y") {
+    if (($survey->aOptions['datestamp'] == "Y") || (($survey->aOptions['datestamp'] != "N") && (SurveysGroupSettings::model()->findByPk(0)->datestamp == "Y"))) {
```

Comment:
This makes not alot of sense - just check for $survey->aOptions['datestamp'] == "Y" as before.

If you are worried that there might be something different than Y/N/I then implement an ActiveRecord rule that keeps an invalid value from ending up in the database, but not validate everywhere else.

Reply:
@c-schmitz yes, that was the exact worry. Removed that part. If we are worried about this, we can create a separate ticket and separate that concern from this bug